tier3/hooks/post_gen_project.py

```diff
@@ -0,0 +1,86 @@
+import subprocess
+
+REPO_NAME = '{{ cookiecutter.project_repo_name }}'
+ORG_NAME = '{{ cookiecutter.project_org }}'
+VISIBILITY = '{{cookiecutter.project_visibility}}'
+DESCRIPTION = '{{cookiecutter.project_description}}'
+CREATE_REPO = '{{cookiecutter.create_repo}}'
+RECEIVE_UPDATES = '{{cookiecutter.receive_updates}}'
+ADD_MAINTAINER = '{{cookiecutter.add_maintainer}}'
+
+def createGithubRepo():
+    subprocess.call(["git", "init", "-b", "main"])
+    subprocess.call(["git", "add", "."])
+    subprocess.call(["git", "commit", "-m", "first commit"])
+    gh_cli_command = [
+        "gh", "repo", "create",
+        f"{ORG_NAME}/{REPO_NAME}",
+        "--source=.",
+        f"--{VISIBILITY}",
+        "--push",
+        f"--description={DESCRIPTION}",
+    ]
+    subprocess.call(gh_cli_command)
+    subprocess.call(["git", "push", "--set-upstream", "origin", "main"])
+
+def addTopic():
+    gh_cli_command = [
+        "gh", "repo", "edit",
+        f"{ORG_NAME}/{REPO_NAME}",
+        "--add-topic=dsacms-tier0",
+    ]
+    subprocess.call(gh_cli_command)
+
+# Helper function for addMaintainer() to get user input of usernames for Maintainer, Approver, and Reviewer
+def get_usernames(role):
+    while True:
+        usernames = input(f"Enter the GitHub usernames of {role} (comma-separated): ").strip()
+        if usernames:
+            return [username.strip() for username in usernames.split(',')]
+        print("Please enter at least one username.")
+
+# Helper function for addMaintainer() to format list of usernames
+def format_usernames(usernames):
+    return "".join(f"- @{username}\n" for username in usernames)
+
+def addMaintainer():
+    maintainers = get_usernames("MAINTAINERS")
+    approvers = get_usernames("APPROVERS")
+    reviewers = get_usernames("REVIEWERS")
+
+    maintainers_file_path = "MAINTAINERS.md"
+
+    with open(maintainers_file_path, "r") as f:
+        lines = f.readlines()
+
+    in_maintainers, in_approvers, in_reviewers = False, False, False
```

Comment:
These variables are probably unneeded, the same logic should work in a more simple way if you did something like this:
```py

if line.strip() == "## Maintainers:" and lines[i+1].strip() == "-":
    lines[i+1] = format_usernames(maintainers)
elif line.strip == "## Approvers:" and lines[i+1].strip() == "-":
    #and so on 
```

Reply:
I implemented this method but since there's a comment right below the Maintainers title, it messes up with the logic that only applies to the Maintainers role list. I'll temporarily re-implement the original code for now.

Reply:
I gotcha, that sounds just like an off by one issue though. I can help if it ends up giving you too much trouble